src/nvim/insexpand.c

```diff
@@ -3291,6 +3299,36 @@ static bool get_next_completion_match(int type, ins_compl_next_state_T *st, pos_
   return found_new_match;
 }
 
+static void get_next_bufname_token(void)
+{
+  int flags = p_ic ? CP_ICASE : 0;
+  char *p;
+
+  FOR_ALL_BUFFERS(b) {
+    if (b->b_p_bl) {
+      char *copy = get_past_head(b->b_sfname);
+      char *original_copy = copy;
+      p = (char *)path_next_component(copy);
+      while (true) {
+        int len = (int)(p - original_copy) - 1;
+        char *dot = strchr(p, '.');
+        if (dot && *p != '.') {
+          *dot = '/';
+        }
+        ins_compl_add(original_copy, len, NULL, NULL, false, NULL, 0,
+                      flags, false);
+        original_copy = p;
+        p = (char *)path_next_component(p);
+        if (*p == NUL) {
```

Comment:
I think we can simplify the code to not need a duplicate `ins_compl_add(...)`. you can remove this if check here, the next iteration of `while (true)` will handle it just fine even if `*p = NUL`. then just check for this condition before calling `path_next_component()` instead.

Reply:
done.